homeassistant/components/coronavirus/sensor.py

```diff
@@ -51,9 +51,14 @@ def available(self):
     def state(self):
         """State of the sensor."""
         if self.country == OPTION_WORLDWIDE:
-            return sum(
-                getattr(case, self.info_type) for case in self.coordinator.data.values()
-            )
+            sum_cases = 0
+            for case in self.coordinator.data.values():
+                value = getattr(case, self.info_type)
+                if value is None:
+                    return None
```

Comment:
return or continue to the next case?
```suggestion
                    continue
```

Reply:
If we continue the different types of sums won't add up evenly unless a country doesn't report any values. But what seems to be the problem is that some countries are reporting None for some types of measurements.

Reply:
Well, the problem with this approach, is that `current` and `recovered` will always be "Unknown" (None) because of this. Continuing would at least use all available data.

Reply:
Yes, but the total data would be incorrect.

Reply:
We should probably exclude the countries completely that have incomplete data from the total data.

Reply:
@MartinHjelmare maybe it would be better to set `None` values to `0`.
This way it's not going to break the total counters.

Reply:
No. We don't know that it's zero. We don't know what the value is so we shouldn't report incorrect data.

The best solution in my opinion is to exclude the countries completely as I've written above.

Reply:
I personally consume the data as "known"/confirmed numbers. Even when skipping a country for just a single stat, the total represented at that point, will still contain the fact of the reported numbers world wide for all countries.

This is how Johns Hopkins reports it itself as well:

![image](https://user-images.githubusercontent.com/195327/85230722-58b5fa00-b3f2-11ea-892d-30ec64a4e03a.png)

Reply:
@frenck was checking exactly the same thing... :)
Sweden, for example, is not even listed in the recovered cased so they just ignore it or do +0 in sum counter.

Reply:
Ok.